server/tests/test_sample_all_ballots.py

```diff
@@ -382,3 +380,186 @@ def test_all_ballots_audit(
+    rv = post_json(
+        client,
+        f"/api/election/{election_id}/jurisdiction/{jurisdiction_ids[0]}/round/{round_1_id}/audit-board",
+        [{"name": "Audit Board #1"}, {"name": "Audit Board #2"}],
+    )
+
+    rv = client.get(
+        f"/api/election/{election_id}/jurisdiction/{jurisdiction_ids[0]}/contest"
+    )
+    contest = json.loads(rv.data)["contests"][0]
+
+    # Record invalid results
+    invalid_results = [
+        (
+            [
+                {
+                    "batchName": "",
+                    "batchType": "Provisional",
+                    "choiceResults": {
+                        choice["id"]: choice["numVotes"] / 4
+                        for choice in contest["choices"]
+                    },
+                }
+            ],
+            "'' is too short",
+        ),
+        (
+            [
+                {
+                    "batchName": None,
+                    "batchType": "Provisional",
+                    "choiceResults": {
+                        choice["id"]: choice["numVotes"] / 4
+                        for choice in contest["choices"]
+                    },
+                }
+            ],
+            "None is not of type 'string'",
+        ),
+        (
+            [
+                {
+                    "batchType": "Provisional",
+                    "choiceResults": {
+                        choice["id"]: choice["numVotes"] / 4
+                        for choice in contest["choices"]
+                    },
+                }
+            ],
+            "'batchName' is a required property",
+        ),
+        (
+            [
+                {
+                    "batchName": "Batch 1",
+                    "batchType": "bad type",
+                    "choiceResults": {
+                        choice["id"]: choice["numVotes"] / 4
+                        for choice in contest["choices"]
+                    },
+                }
+            ],
+            "'bad type' is not one of ['Absentee By Mail', 'Advance', 'Election Day', 'Provisional', 'Other']",
+        ),
+        (
+            [
+                {
+                    "batchName": "Batch 1",
+                    "batchType": "bad type",
+                    "choiceResults": {
+                        choice["id"]: choice["numVotes"] / 4
+                        for choice in contest["choices"]
+                    },
+                }
+            ],
+            "'bad type' is not one of ['Absentee By Mail', 'Advance', 'Election Day', 'Provisional', 'Other']",
+        ),
+        (
+            [
+                {
+                    "batchName": "Batch 1",
+                    "batchType": "Provisional",
+                    "choiceResults": {
+                        choice["id"]: choice["numVotes"] / 4
+                        for choice in contest["choices"][:1]
+                    },
+                }
+            ],
+            "Invalid choice ids for batch Batch 1",
```

Comment:
Do we want a more informative error here? "All choices must have total (even if it is 0)"?

Reply:
users should never see this - the frontend doesnt allow it. it's just defense